flake.nix

```diff
@@ -45,6 +45,14 @@
         }).nixos.manual.x86_64-linux;
       };
 
+      # The "legacy" in `legacyPackages` doesn't imply that the packages exposed
+      # through this attribute are "legacy" packages. Instead, `legacyPackages`
+      # is used here as a substitute for `packages` because enumerating the
+      # entire list of packages in Nixpkgs is quite computationally expensive
+      # due to the sheer number of packages, which in turn makes an operation
+      # like `nix flake show nixpkgs` unusably slow. The Nix CLI recognizes this
+      # substitute name only in Nixpkgs; it doesn't have the same meaning in
```

Comment:
Is this true? I don't see any Nixpkgs-specific special-casing in Nix (but maybe I'm not looking hard enough).

I have a `legacyPackages` attribute in my NixOS flake, and it has the same behavior:

```
$ nix flake show
git+file:///home/vin/flake                                                             
├───colmena: unknown                                                                   
├───inputs: unknown                                                                    
├───legacyPackages                                                                     
│   └───x86_64-linux omitted (use '--legacy' to show)                                  
├───lib: unknown                                                                       
└───nixosConfigurations                                                                
    ├───bootstrap: NixOS configuration                                                 
    ├───scadrial: NixOS configuration                                                  
    └───scar: NixOS configuration                                                      

```

```
$ nix flake show nixpkgs                                                                                                                                                                                                                    
path:/nix/store/qszh53md64g0jxfb9akahgs50wsnmpiz-source?lastModified=1664281702&rev=7e52b35fe98481a279d89f9c145f8076d049d2b9                                                                                                                    
├───checks                                                                                                                                                                                                                                      
│   └───x86_64-linux                                                                                                                                                                                                                            
│       └───tarball: derivation 'nixpkgs-tarball-22.11pre20220927.7e52b35'                                                                                                                                                                      
├───htmlDocs: unknown                                                                                                                                                                                                                           
├───legacyPackages                                                                                                                                                                                                                              
│   ├───aarch64-darwin omitted (use '--legacy' to show)                                                                                                                                                                                         
│   ├───aarch64-linux omitted (use '--legacy' to show)                                                                                                                                                                                          
│   ├───armv5tel-linux omitted (use '--legacy' to show)                                                                                                                                                                                         
│   ├───armv6l-linux omitted (use '--legacy' to show)                                                                                                                                                                                           
│   ├───armv7l-linux omitted (use '--legacy' to show)                                                                                                                                                                                           
│   ├───i686-linux omitted (use '--legacy' to show)                                                                                                                                                                                             
│   ├───mipsel-linux omitted (use '--legacy' to show)                                                                                                                                                                                           
│   ├───powerpc64le-linux omitted (use '--legacy' to show)                                                                                                                                                                                      
│   ├───riscv64-linux omitted (use '--legacy' to show)                                                                                                                                                                                          
│   ├───x86_64-darwin omitted (use '--legacy' to show)                                                                                                                                                                                          
│   └───x86_64-linux omitted (use '--legacy' to show)                                                                                                                                                                                           
├───lib: unknown                                                                                                                                                                                                                                
└───nixosModules                                                                                                                                                                                                                                
    └───notDetected: NixOS module                                                                                                                                                                                                               

```

Reply:
The wiki suggests that it's specific to Nixpkgs: https://nixos.wiki/wiki/Flakes. But good to know that it's indeed _not_ hard wired.